start_esp/start_esp.py

```diff
@@ -150,34 +150,56 @@ def start_nginx(nginx, nginx_conf):
 
 def fetch_service_config(args, service_config):
     try:
-        # Fetch service config
-        if args.service_config_url is None:
+        if args.service_config_url is not None:
+            service_mgmt_url = args.service_config_url
+        else:
+            # fetch service name, if not specified
             if args.service is None:
-                logging.info("Fetching the service name from the metadata service")
+                logging.info(
+                    "Fetching the service name from the metadata service")
                 args.service = fetch.fetch_service_name(args.metadata)
 
+            # if service name is not specified, display error message and exit
+            if args.service is None:
+                logging.error("Unable to get service name");
+                sys.exit(3)
+
+            # fetch service config ID, if not specified
             if args.version is None:
-                logging.info("Fetching the service config ID from the metadata service")
+                logging.info("Fetching the service config ID "\
+                             "from the metadata service")
                 args.version = fetch.fetch_service_config_id(args.metadata)
 
-            service_mgmt_url = SERVICE_MGMT_URL_TEMPLATE.format(args.service,
-                                                                args.version)
-        else:
-            service_mgmt_url = args.service_config_url
+            # Get the access token
+            if args.service_account_key is None:
+                logging.info("Fetching an access token from the metadata service")
+                token = fetch.fetch_access_token(args.metadata)
+            else:
+                token = fetch.make_access_token(args.service_account_key)
 
-        # Get the access token
-        if args.service_account_key is None:
-            logging.info("Fetching an access token from the metadata service")
-            token = fetch.fetch_access_token(args.metadata)
-        else:
-            token = fetch.make_access_token(args.service_account_key)
+            # Fetch api version from latest successful rollouts
+            if args.version is None:
+                logging.info(
+                    "Fetching the service config ID from the rollouts service")
+                rollout = fetch.fetch_latest_rollout(args.service, token)
+                # For phase 1, only one config ID in each rollout
+                for version, percentage in rollout.iteritems():
```

Comment:
Good point,   I think the CLI to create a rollout with multiple versions is NOT ready yet.  but the API is there,  people can use API directly to create one.   

ESP should support loading multiple version rollout even in phase 1.  We can do this in a separate PR.  That is what ESP should do:

1) change server_config.proto to allow passing multiple file full paths of service config with percentage, ESP will load them in, and use them with proper traffic percentage.   This can be "t" tested too.

2) we can use server_config to pass many of ESP settings from start_esp to ESP,   bypass Nginx config.